### ISSUE TITLE
interaction between AOT compiled modules gives out of bounds memory access, Code: 0x88 or unreachable

### ISSUE THREAD

**Issue by loredanacirstea** [WasmEdge/WasmEdge#2627]
## Description

I am using Wasmedge v0.11.2 & `wasmedge-go` to run wasm modules that import a suite of host functions and the wasm modules are dependent on one another. e.g. I have two `wasmA` and `wasmB` modules, where `wasmA` calls `wasmB` through host functions:
- execute exported function `wasmA.main` -> execute wasmA imported host function `externalCall` -> `externalCall` instantiates wasmB and calls `wasmB.main` function, gets the result and releases the wasmB resources -> wasmA receives result from wasmB and continues execution.

When I run the wasm modules in the interpreted mode, everything works. When I run them as AOT compiled, I get these errors:

- if I call `Release()` on wasmB's VM (created with`wasmedge.NewVM()`) at the end of wasmB execution, I get:
```sh
[error] execution failed: out of bounds memory access, Code: 0x88
[error]     When executing function name: "main"
```
- if I don't release wasmB's VM, I get:
```sh
[error] execution failed: unreachable, Code: 0x89
[error]     When executing function name: "main"
```

Note that wasmA and wasmB have no shared resources. I use a separate `wasmedge.NewVM()` for each.

These errors appear after wasmA correctly receives the result from wasmB, immediately after the `externalCall` host function returns the result and wasmA resumes execution. If I don't execute the `wasmB.main` call, there is no error and the execution of `wasmA` continues normally.

It seems the issue appears only when an AOT compiled wasm makes this `externalCall` to another wasm module. If the call is made from an interpreted wasm module to an AOT compiled wasm, the execution is successful.

Is there any known issue of this sort, for AOT compilation? Why would the execution of wasmB influence the state of wasmA?

## Environment

- Hardware Architecture: (e.g. x86\_64 or aarch64): darwin x86_64
- Operating system: (e.g. Ubuntu 20.04 or CentOS 7.6): MacOS
- WasmEdge 0.11.2

## Steps to Reproduce

I cannot provide the source code I am working with. I tried to reproduce the bug with a very simple example, but failed.
I wanted to try this on v 0.12.1, but could not make it work with the current wasmedge-go v0.12.0.

**Comment by q82419** [comment]
It may be fixed in `0.13.0`.
Just comment that we can try the `0.13.0` in these days.

**Comment by hydai** [comment]
Since the reporter is not going to share the WASM and this error cannot be reproduced, there is no way to figure out the reason and fix it. Closing this.
